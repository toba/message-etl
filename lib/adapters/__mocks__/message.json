{
   "participants": [
      {
         "name": "Other Person"
      },
      {
         "name": "Self Person"
      }
   ],
   "messages": [
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539713573381,
         "content": "Oh, I know what that means. Ok",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539713210175,
         "content": "Maybe I translated wrong. Full message: \"Its 3160 Lewis River Rd Woodland Wa 98674 for know it will be changing when the new houses and road goes in don't when that will happen.\"",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539713187186,
         "content": "What do you mean not in the new place?",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539713157344,
         "content": "Jessica: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539553590548,
         "content": "You sent a video.",
         "videos": [
            {
               "uri": "messages/OtherPerson_ea77ad6afb/videos/video1539553590_336614783757508.mp4",
               "creation_timestamp": 1539553586,
               "thumbnail": {
                  "uri": "messages/OtherPerson_ea77ad6afb/videos/thumbnails/42075454_2168518689894738_1310336861426507451_n.jpg"
               }
            }
         ],
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539712398807,
         "content": "And he spelled \"your\" wrong",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539465166225,
         "content": "You: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539712382725,
         "content": "Scott says the address is still 3160 Lewis River Rd. Not in the new place yet.",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539705094711,
         "content": "Rough",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539704752058,
         "content": "Went to make latte for first time ever here and we're out of milk. What a dump.",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539657694856,
         "content": "Jessica: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657676405,
         "content": "Was on perch a while",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657668688,
         "content": "Mellow",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657665741,
         "content": "Seems to be",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539657639664,
         "content": "She feeling okay?",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657593609,
         "content": "Cats new hiding spot",
         "reactions": [
            {
               "reaction": "\u00f0\u009f\u0098\u0086",
               "actor": "Other Person"
            }
         ],
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657585171,
         "content": "You sent a photo.",
         "photos": [
            {
               "uri": "messages/JessicaAbbott_ea77ad6afb/photos/44126872_2446710928889289_825560690738593792_n_2446710902222625.jpg",
               "creation_timestamp": 1539657584
            }
         ],
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539653084097,
         "content": "So very sad",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539652968444,
         "content": "Felt so sad going in his room to turn off the light, looking around",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539649902196,
         "content": "You: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539649893421,
         "content": "Oh, I know it was 3160. I'll find out",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539649832938,
         "content": "Meant as more of a question :)",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539649769629,
         "content": "Thanks",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539649670977,
         "content": "3170 Lewis River Rd\nWoodland, WA 98674",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539649669514,
         "content": "The address I have is",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539643699846,
         "content": "You: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539643632522,
         "content": "I'll be by to get Hunter around 5:30",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539643575006,
         "content": "Waiting in line for cat",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539633414962,
         "content": "yeah ...",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539633395761,
         "content": "no clue on the xbox, sold is probably a good bet. He can deal with Scott on that one",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539633266739,
         "content": "didn't see xbox anywhere. i guess he sold it? or I put it somewhere I totally forgot? but i don't think so. just had it in the office, locked.",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539633220098,
         "content": "checked his room. it seemed clean in the sense that there aren't clothes on the floor anymore.",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539633194653,
         "content": "If he's back by 4:30 and his room is clean and he's totally packed",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539632575070,
         "content": "Hunter wanting to go visit friends. I said to check with you.",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1371924211989,
         "content": "This is my route\nhttps://dl.dropboxusercontent.com/u/33199511/lava-mountain-loop.jpg",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1370924843625,
         "content": "Good night, sweet dreams reality is obviously not your forte.",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1370872998841,
         "content": "SPOILED monster.\n",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1370793302848,
         "content": "That is hilarious, we are all laughing. Hitting the road now",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1298522821000,
         "content": "Do you ever have those moments where somebody is so mind shatteringly selfish and horrible that it may take you a few days to even  put the right words together, because it is so unfathomable for most of the population to be that way?? Yet they think they are logical and normal?? Even though everybody around them has to put effort into tolerating being around them??? No matter what damage they cause in their wake they will NEVER change anything or say they are sorry???",
         "type": "Generic"
      }
   ],
   "title": "Thread Title",
   "is_still_participant": true,
   "thread_type": "Regular",
   "thread_path": "OtherPerson_ea77ad6afb"
}
