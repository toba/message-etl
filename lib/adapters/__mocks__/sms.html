<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>First Last-Hyphen</title>
<style type="text/css">
          /* Copyright 2011 Google Inc.  All Rights Reserved. */

body {
  font-size: 13px;
  font-family: Arial, Helvectica, sans-serif;
}

a {
  color: #00c;
}

a:hover {
  text-decoration: underline;
}

.tags,
.hChatLog,
.noteContainer {
  margin: 0 auto;
  width: 750px;
  min-width: 750px;
}

.message {
  max-width: 640px;
}

/* The following three style rules exist to override default behavior in the
 * user-agent stylesheet used by common browsers such as Chrome. */
cite {
  font-style: normal;
}

q::before {
  content: "";
}

q::after {
  content: "";
}

.tags,
.noteContainer {
  margin-top: 13px;
}

.participants {
  margin-bottom: 13px;
}

        </style></head>
<body><div class="hChatLog hfeed">
<div class="message"><abbr class="dt" title="2013-06-17T20:46:00.861-04:00">Jun 17, 2013, 8:46:00 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>Flight 2244, 15 minutes delayed, also had to check in bag because Carolyn bought Kayla and I stuff from bath &amp; body works</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T20:48:20.247-04:00">Jun 17, 2013, 8:48:20 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12085508050"><abbr class="fn" title="">Me</abbr></a></cite>:
<q>Have good flying</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T22:47:28.109-04:00">Jun 17, 2013, 10:47:28 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>Landed</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T22:47:58.180-04:00">Jun 17, 2013, 10:47:58 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12085508050"><abbr class="fn" title="">Me</abbr></a></cite>:
<q>We&#39;ll drive up the hill and await your text</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T22:48:46.831-04:00">Jun 17, 2013, 10:48:46 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>Ok</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T23:00:17.481-04:00">Jun 17, 2013, 11:00:17 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>Outside</q>
</div> <div class="message"><abbr class="dt" title="2013-06-22T15:08:16.526-04:00">Jun 22, 2013, 3:08:16 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12085508050"><abbr class="fn" title="">Me</abbr></a></cite>:
<q>At Bonneville point</q>
</div> <div class="message"><abbr class="dt" title="2013-06-22T15:09:13.122-04:00">Jun 22, 2013, 3:09:13 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>So far so good, nice day for a ride</q>
</div> <div class="message"><abbr class="dt" title="2013-06-26T19:51:05.573-04:00">Jun 26, 2013, 7:51:05 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12085508050"><abbr class="fn" title="">Me</abbr></a></cite>:
<q>Superman? Home in 15</q>
</div></div>

<div class="tags">Labels:
<a rel="tag" href="http://www.google.com/voice#inbox">Inbox</a>, <a rel="tag" href="http://www.google.com/voice#sms">Text</a></div></body></html>