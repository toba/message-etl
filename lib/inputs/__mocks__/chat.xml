<?xml version="1.0"?>
<?xml-stylesheet type='text/xsl' href='MessageLog.xsl'?>
<Log LogonName="other@trailimage.com" FirstSessionID="1" LastSessionID="26">
   <Message Date="6/24/2003" Time="10:24:17 AM" DateTime="2003-06-24T16:24:17.396Z" SessionID="1">
      <From>
         <User LogonName="other@trailimage.com" FriendlyName="Other Person"/>
      </From>
      <To>
         <User LogonName="self@trailimage.com" FriendlyName="Self Person"/>
      </To>
      <Text Style="font-family:Microsoft Sans Serif; font-weight:bold; font-style:italic; color:#008080; ">let me guess, working hard again?</Text>
   </Message>
   <Message Date="6/24/2003" Time="10:24:42 AM" DateTime="2003-06-24T16:24:42.102Z" SessionID="1">
      <From>
         <User LogonName="self@trailimage.com" FriendlyName="Self Person"/>
      </From>
      <To>
         <User LogonName="other@trailimage.com" FriendlyName="Other Person"/>
      </To>
      <Text Style="font-family:Tahoma; color:#800080; ">yes</Text>
   </Message>
   <Message Date="6/24/2003" Time="10:25:54 AM" DateTime="2003-06-24T16:25:54.216Z" SessionID="1">
      <From>
         <User LogonName="other@trailimage.com" FriendlyName="Other Person"/>
      </From>
      <To>
         <User LogonName="sunflower77h@hotmail.com" FriendlyName="butterfly"/>
         <User LogonName="self@trailimage.com" FriendlyName="Jason Abbott"/>
         <User LogonName="momofhailey@hotmail.com" FriendlyName="don't you wish you were floating the tubes!"/>
      </To>
      <Text Style="font-family:Microsoft Sans Serif; font-weight:bold; font-style:italic; color:#008080; ">Laura wearing you out yet?</Text>
   </Message>
   <Message Date="6/24/2003" Time="10:29:06 AM" DateTime="2003-06-24T16:29:06.953Z" SessionID="1">
      <From>
         <User LogonName="self@trailimage.com" FriendlyName="Self Person"/>
      </From>
      <To>
         <User LogonName="other@trailimage.com" FriendlyName="Other Person"/>
      </To>
      <Text Style="font-family:Tahoma; color:#800080; ">sorry, been on and still on the phone</Text>
   </Message>
</Log>