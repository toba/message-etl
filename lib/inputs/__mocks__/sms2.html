<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>First Last-Hyphen</title>
<style type="text/css">
          /* Copyright 2011 Google Inc.  All Rights Reserved. */

body {
  font-size: 13px;
  font-family: Arial, Helvectica, sans-serif;
}

a {
  color: #00c;
}

a:hover {
  text-decoration: underline;
}

.tags,
.hChatLog,
.noteContainer {
  margin: 0 auto;
  width: 750px;
  min-width: 750px;
}

.message {
  max-width: 640px;
}

/* The following three style rules exist to override default behavior in the
 * user-agent stylesheet used by common browsers such as Chrome. */
cite {
  font-style: normal;
}

q::before {
  content: "";
}

q::after {
  content: "";
}

.tags,
.noteContainer {
  margin-top: 13px;
}

.participants {
  margin-bottom: 13px;
}

        </style></head>
<body><div class="hChatLog hfeed">
<div class="message"><abbr class="dt" title="2012-09-25T19:19:11.272-04:00">Sep 25, 2012, 7:19:11 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>At park for a bit, signed papers</q>
</div></div>

<div class="tags">Labels:
<a rel="tag" href="http://www.google.com/voice#inbox">Inbox</a>, <a rel="tag" href="http://www.google.com/voice#sms">Text</a></div></body></html>