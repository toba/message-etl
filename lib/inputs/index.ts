export { facebookMessenger as fb } from './messenger';
export { googleVoice as gv } from './google-voice';
export { mailbox as mb } from './mbox';
export { microsoftChat as chat } from './chat';
export { passport } from './passport';
