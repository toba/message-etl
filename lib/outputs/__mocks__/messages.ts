import { lipsum } from '@toba/test';
import { Message, Relation, Source } from '../../types';

export const messages: Message[] = [
   {
      from: Relation.Self,
      on: new Date(2014, 5, 12),
      text: lipsum,
      source: Source.FacebookMessenger
   }
];
