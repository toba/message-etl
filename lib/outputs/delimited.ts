import { Message, Writer } from '../types';
import { people } from '../config';

//`${m.on.getMonth() + 1}/${m.on.getDate()}/${m.on.getFullYear()}`

export const serialize = (m: Message): string =>
   [m.on.toISOString(), people[m.from].name, m.text].join('|');

export const delimited: Writer = {
   write: (messages: Message[]) => messages.map(serialize).join('\n'),
   fileName: 'messages.csv',
   serialize
};
