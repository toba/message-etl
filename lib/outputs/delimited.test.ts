import '@toba/test';
import { readFileText } from '@toba/test';
import { fb } from '../inputs/';
import { delimited as d } from './delimited';
import { messages } from './__mocks__/messages';

jest.mock('../config/index');

test('serializes message', () => {
   expect(d.serialize(messages[0])).toMatchSnapshot();
});

test('generates delimited file', async () => {
   const text = await readFileText(
      `${__dirname}/../inputs/__mocks__/message.json`
   );
   const messages = fb.process(text);
   const file = d.write(messages);

   expect(file).toMatchSnapshot();
});
