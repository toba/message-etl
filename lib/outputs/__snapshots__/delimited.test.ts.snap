// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates delimited file 1`] = `
"2018-10-16T18:12:53.381Z|Other|Oh, I know what that means. Ok
2018-10-16T18:06:50.175Z|Self|Maybe I translated wrong. Full message: \\"Its 3160 Lewis River Rd Woodland Wa 98674 for know it will be changing when the new houses and road goes in don't when that will happen.\\"
2018-10-16T18:06:27.186Z|Other|What do you mean not in the new place?
2018-10-16T18:05:57.344Z|Other|Jessica: 󰀀
2018-10-14T21:46:30.548Z|Self|You sent a video.
2018-10-16T17:53:18.807Z|Self|And he spelled \\"your\\" wrong
2018-10-13T21:12:46.225Z|Self|You: 󰀀
2018-10-16T17:53:02.725Z|Self|Scott says the address is still 3160 Lewis River Rd. Not in the new place yet.
2018-10-16T15:51:34.711Z|Other|Rough
2018-10-16T15:45:52.058Z|Self|Went to make latte for first time ever here and we're out of milk. What a dump.
2018-10-16T02:41:34.856Z|Other|Jessica: 󰀀
2018-10-16T02:41:16.405Z|Self|Was on perch a while
2018-10-16T02:41:08.688Z|Self|Mellow
2018-10-16T02:41:05.741Z|Self|Seems to be
2018-10-16T02:40:39.664Z|Other|She feeling okay?
2018-10-16T02:39:53.609Z|Self|Cats new hiding spot
2018-10-16T02:39:45.171Z|Self|You sent a photo.
2018-10-16T01:24:44.097Z|Other|So very sad
2018-10-16T01:22:48.444Z|Self|Felt so sad going in his room to turn off the light, looking around
2018-10-16T00:31:42.196Z|Self|You: 󰀀
2018-10-16T00:31:33.421Z|Other|Oh, I know it was 3160. I'll find out
2018-10-16T00:30:32.938Z|Self|Meant as more of a question :)
2018-10-16T00:29:29.629Z|Other|Thanks
2018-10-16T00:27:50.977Z|Self|3170 Lewis River Rd
Woodland, WA 98674
2018-10-16T00:27:49.514Z|Self|The address I have is
2018-10-15T22:48:19.846Z|Self|You: 󰀀
2018-10-15T22:47:12.522Z|Other|I'll be by to get Hunter around 5:30
2018-10-15T22:46:15.006Z|Self|Waiting in line for cat
2018-10-15T19:56:54.962Z|Self|yeah ...
2018-10-15T19:56:35.761Z|Other|no clue on the xbox, sold is probably a good bet. He can deal with Scott on that one
2018-10-15T19:54:26.739Z|Self|didn't see xbox anywhere. i guess he sold it? or I put it somewhere I totally forgot? but i don't think so. just had it in the office, locked.
2018-10-15T19:53:40.098Z|Self|checked his room. it seemed clean in the sense that there aren't clothes on the floor anymore.
2018-10-15T19:53:14.653Z|Other|If he's back by 4:30 and his room is clean and he's totally packed
2018-10-15T19:42:55.070Z|Self|Hunter wanting to go visit friends. I said to check with you.
2013-06-22T18:03:31.989Z|Self|This is my route
https://dl.dropboxusercontent.com/u/33199511/lava-mountain-loop.jpg
2013-06-11T04:27:23.625Z|Other|Good night, sweet dreams reality is obviously not your forte.
2013-06-10T14:03:18.841Z|Other|SPOILED monster.
2013-06-09T15:55:02.848Z|Other|That is hilarious, we are all laughing. Hitting the road now
2011-02-24T04:47:01.000Z|Other|Do you ever have those moments where somebody is so mind shatteringly selfish and horrible that it may take you a few days to even  put the right words together, because it is so unfathomable for most of the population to be that way?? Yet they think they are logical and normal?? Even though everybody around them has to put effort into tolerating being around them??? No matter what damage they cause in their wake they will NEVER change anything or say they are sorry???"
`;

exports[`serializes message 1`] = `"2014-06-12T06:00:00.000Z|Self|Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."`;
